src/shadowbox/server/shared_metrics.spec.ts

```diff
@@ -50,212 +71,160 @@ describe('OutlineSharedMetricsPublisher', () => {
       expect(publisher.isSharingEnabled()).toBeFalsy();
       expect(serverConfig.mostRecentWrite.metricsEnabled).toBeFalsy();
     });
+
     it('Reads from config', () => {
-      const serverConfig = new InMemoryConfig<ServerConfigJson>({metricsEnabled: true});
-      const publisher = new OutlineSharedMetricsPublisher(
-        new ManualClock(),
-        serverConfig,
-        null,
-        null,
-        null
-      );
+      serverConfig.data().metricsEnabled = true;
+
       expect(publisher.isSharingEnabled()).toBeTruthy();
     });
   });
-  describe('Metrics Reporting', () => {
-    it('reports server usage metrics correctly', async () => {
-      const clock = new ManualClock();
-      let startTime = clock.nowMs;
-      const serverConfig = new InMemoryConfig<ServerConfigJson>({serverId: 'server-id'});
-      const usageMetrics = new ManualUsageMetrics();
-      const metricsCollector = new FakeMetricsCollector();
-      const publisher = new OutlineSharedMetricsPublisher(
-        clock,
-        serverConfig,
-        null,
-        usageMetrics,
-        metricsCollector
-      );
 
+  describe('reporting', () => {
+    beforeEach(() => {
       publisher.startSharing();
-      usageMetrics.countryUsage = [
-        {country: 'AA', inboundBytes: 11},
-        {country: 'BB', inboundBytes: 11},
-        {country: 'CC', inboundBytes: 22},
-        {country: 'AA', inboundBytes: 33},
-        {country: 'DD', inboundBytes: 33},
-      ];
-
-      clock.nowMs += 60 * 60 * 1000;
-      await clock.runCallbacks();
-      expect(metricsCollector.collectedServerUsageReport).toEqual({
-        serverId: 'server-id',
-        startUtcMs: startTime,
-        endUtcMs: clock.nowMs,
-        userReports: [
-          {bytesTransferred: 11, countries: ['AA']},
-          {bytesTransferred: 11, countries: ['BB']},
-          {bytesTransferred: 22, countries: ['CC']},
-          {bytesTransferred: 33, countries: ['AA']},
-          {bytesTransferred: 33, countries: ['DD']},
-        ],
+    });
+
+    afterEach(() => {
+      publisher.stopSharing();
+    });
+
+    describe('for server usage', () => {
+      it('is sending correct reports', async () => {
+        usageMetrics.countryUsage = [
+          {country: 'AA', inboundBytes: 11},
+          {country: 'BB', inboundBytes: 11},
+          {country: 'CC', inboundBytes: 22},
+          {country: 'AA', inboundBytes: 33},
+          {country: 'DD', inboundBytes: 33},
+        ];
+        clock.nowMs += 60 * 60 * 1000;
+
+        await clock.runCallbacks();
+
+        expect(metricsCollector.collectServerUsageMetrics).toHaveBeenCalledOnceWith({
```

Comment:
Prefer a fake implementation (as we had before) than interaction verification: [go/choose-test-double](http://go/choose-test-double)

go/choose-test-double#prefer-fakes